File: Source/DotNET/Backend/Features/FeatureToggles.cs
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Dolittle.Vanir.Backend.Features
{
    /// <summary>
    /// Represents an implementation of <see cref="IFeatureToggles"/>
    /// </summary>
    public class FeatureToggles : IFeatureToggles
    {
        readonly Features _features;

        /// <summary>
        /// Initializes a new instance of <see cref="FeatureToggles"/>.
        /// </summary>
        /// <param name="provider"><see cref="IFeaturesProvider"/> for providing features.</param>
        public FeatureToggles(IFeaturesProvider provider)
        {
            _features = provider.Provide();
        }

        /// <inheritdoc/>
        public bool IsOn(string feature)
        {
            if (!_features.ContainsKey(feature)) return false;
            return _features[feature].IsOn;
        }
    }
}

File: Source/typescript/features/FeatureToggles.ts
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { IFeatureToggles } from './IFeatureToggles';

/**
 * Represents an implementation of {@link IFeatureToggles}
 */

export class FeatureToggles implements IFeatureToggles {

    /** @inheritdoc */
    isOn(feature: string): boolean {
        throw new Error('Method not implemented.');
    }
}
